### PR TITLE
DEV: pull remove-open-orders logic into its own method

### PR DESCRIPTION
**Pull request by jbredeche**
And test it.

### CODE REVIEW

File: zipline/finance/blotter.py [hunk 1]
```diff
@@ -329,20 +329,23 @@ def get_transactions(self, bar_data):
...
+
+    def prune_orders(self, closed_orders):
         # remove all closed orders from our open_orders dict
         for order in closed_orders:
             sid = order.sid
             try:
                 sid_orders = self.open_orders[sid]
                 sid_orders.remove(order)
-            except KeyError:
+            except ValueError:
```
Comment (richafrank) [inline zipline/finance/blotter.py]: When do we expect this case?  Also, could we move the slice statement out of the try, so it's clearer that that isn't generating the error?

File: zipline/finance/blotter.py [hunk 2]
```diff
@@ -329,20 +329,23 @@ def get_transactions(self, bar_data):
                     if not order.open:
                         closed_orders.append(order)
 
+        self.prune_orders(closed_orders)
```
Comment (richafrank) [inline zipline/finance/blotter.py]: Since this was a refactoring to `get_transactions`, should we have test coverage for that method?

Comment (jbredeche) [inline zipline/finance/blotter.py]: ideally, yes.  I don't have time right now.

File: zipline/finance/blotter.py (hunk 1 above)
Comment (jbredeche) [inline zipline/finance/blotter.py]: I'll add a comment, but we expect this in a downstream subclass that is connected to a broker, and the broker is canceling orders.  There's some scenario there that causes us to try to remove an order that we've already removed locally (probably due to EOD cancel policy on our own end), and thus is happening.

Either way, we're trying to remove something that is already removed, so we shouldn't blow up on it.

Comment (richafrank) [inline zipline/finance/blotter.py]: > Either way, we're trying to remove something that is already removed, so we shouldn't blow up on it.

That's fair.  Just wondering if there's some bad state tracking elsewhere, in that we're trying to remove it at all, and whether that should be fixed instead, as opposed to fixing all those downstream of it.

File: zipline/finance/blotter.py [hunk 3]
```diff
@@ -329,20 +332,29 @@ def get_transactions(self, bar_data):
...
+
+    def prune_orders(self, closed_orders):
+        """
+        Removes all given orders from the blotter's open_orders list.
+
+        Parameters
+        ----------
+
+        :param closed_orders:
+        :return:
```
Comment (richafrank) [inline zipline/finance/blotter.py]: If we're not filling these out, let's just remove 'em.

**Comment by richafrank** [comment]
LGTM!  Just mentioned the docstring.

**Pull request merged**